plugins/draco/io/DracoReader.cpp

```diff
@@ -0,0 +1,363 @@
+        case DT_INT8:
+            return Dimension::Type::Signed8;
+        case DT_UINT8:
+            return Dimension::Type::Unsigned8;
+        case DT_INT16:
+            return Dimension::Type::Signed16;
+        case DT_UINT16:
+            return Dimension::Type::Unsigned16;
+        case DT_INT32:
+            return Dimension::Type::Signed32;
+        case DT_UINT32:
+            return Dimension::Type::Unsigned32;
+        case DT_INT64:
+            return Dimension::Type::Signed64;
+        case DT_UINT64:
+            return Dimension::Type::Unsigned64;
+        case DT_FLOAT32:
+            return Dimension::Type::Float;
+        case DT_FLOAT64:
+            return Dimension::Type::Double;
+
+        default:
+            // Not supported draco domain types
+            throw pdal_error("Invalid Dim type from Draco");
+    }
+}
+
+void DracoReader::addArgs(ProgramArgs& args)
+{
+}
+
+void DracoReader::prepared(PointTableRef table)
+{
+    if (m_filename.empty())
+        throwError("Required argument 'filename' (Draco array name) "
+            "not provided.");
+}
+
+
+void DracoReader::initialize()
+{
+    if (!FileUtils::fileExists(m_filename))
+        throwError("File '" + m_filename + "' does not exist");
+
+    m_istreamPtr = Utils::openFile(m_filename, true);
+    if (!m_istreamPtr)
+        throwError("Unable to open file '" + m_filename + "' ");
+
+    m_data.assign(std::istreambuf_iterator<char>(*m_istreamPtr),
+                std::istreambuf_iterator<char>());
+    Utils::closeFile(m_istreamPtr);
+
+    m_draco_buffer.Init(m_data.data(), m_data.size());
+
+    draco::Decoder decoder;
+    auto geom_status = draco::Decoder::GetEncodedGeometryType(&m_draco_buffer);
+    if (!geom_status.ok())
+    {
+        return throwError(geom_status.status().error_msg());
+    }
+
+    const draco::EncodedGeometryType geom_type = geom_status.value();
+
+    draco::Decoder::GetEncodedGeometryType(&m_draco_buffer);
+
+    draco::StatusOr<std::unique_ptr<draco::PointCloud>> pc_status = decoder.DecodePointCloudFromBuffer(&m_draco_buffer);
+    if (!pc_status.ok())
+    {
+        return throwError(pc_status.status().error_msg());
+    }
+
+    m_pc = std::move(pc_status).value();
+
+    m_count = m_pc->num_points();
+}
+
+
+void DracoReader::addDimensions(PointLayoutPtr layout)
+{
+    using namespace draco;
+
+    log()->get(LogLevel::Debug) << "draco pc num_attributes: "
+                                << m_pc->num_attributes() << std::endl;;
+
+    const GeometryMetadata *metadata = m_pc->GetMetadata();
+    for (int i=0; i < m_pc->num_attributes(); ++i)
+    {
+        const PointAttribute* attr = m_pc->GetAttributeByUniqueId(i);
+        const AttributeMetadata* attr_metadata = m_pc->GetAttributeMetadataByAttributeId(i);
+        DataType dt = attr->data_type();
+        GeometryAttribute::Type at = attr->attribute_type();
+
+        int8_t nc = attr->num_components();
+        log()->get(LogLevel::Debug) << "named component: "
```

Comment:
Tidy up the logs to output useful stuff to someone trying to figure out what's in their Draco file.